frontend/packages/knative-plugin/src/utils/fetch-dynamic-eventsources-utils.ts

```diff
@@ -52,8 +51,8 @@ export const fetchEventSourcesCrd = async () => {
             kind,
             plural,
             id: singular,
-            label,
-            labelPlural: `${label}s`,
+            label: kind,
+            labelPlural: `${kind}s`,
```

Comment:
Hmm... there isn't much we can do here I don't think. But I also don't think slapping `s` on the end of words is good English syntax. For instance `...Class` cannot get `s` as a plural state.

I know this isn't this PR work @debsmita1 -- just making the comment because I see it @invincibleJai FYI.

Reply:
@debsmita1 I think for translation we should follow this pattern i.e add `labelKey` and `labelPluralKey` and use the same where applicable for translations. For plural, I don't think we are saving much let's type the whole string.

https://github.com/openshift/console/blob/master/frontend/packages/pipelines-plugin/src/models/pipelines.ts#L7-L23

@andrewballantyne let us know if the above make sense